File: Data Structures/Graphs/Graph.py
class Edge:
  def __init__(self, fr, to, wt = 1):
    self.fr = fr
    self.to = to
    self.wt = wt

class Graph:
  def __init__(self, edgelist = None):
    self.adjacencyList = dict()
    if edgelist == None: return
    for edge in edgelist:
      self.addNode(edge.fr)
      self.addNode(edge.to)
      self.addEdge(edge)

  def addNode(self, node):
    if node in self.adjacencyList: return
    self.adjacencyList[node] = set()

  def addEdge(self, edge):
    self.adjacencyList[edge.fr].add(edge)

  def PrintAdjacencyList(self):
    for node in self.adjacencyList:
      print(node, sep='', end=' -> ')
      for edge in self.adjacencyList[node]:
        print(f"{edge.to}[{edge.wt}]", sep='', end=' -> ')
      print("X")

edgelist = [
  Edge(1, 2),
  Edge(1, 3),
  Edge(1, 4),
  Edge(2, 3),
  Edge(3, 4),
  Edge(3, 5),
  Edge(4, 5),
]

Graph(edgelist).PrintAdjacencyList()

File: Data Structures/Graphs/Graph.cpp
#include <iostream>
#include <vector>
#include <map>

struct Edge
{
  int fr;
  int to;
  int wt;

  Edge(int fr, int to, int wt = 1)
    : fr(fr), to(to), wt(wt) { }
};

struct Graph
{
private:
  std::map<int, std::vector<std::pair<int, int>>> adjacencyList;

public:
  Graph() { }
  Graph(std::vector<Edge>& edgelist)
  {
    for (Edge edge : edgelist)
    {
      AddNode(edge.wt);
      AddNode(edge.to);
      AddEdge(edge);
    }
  }

  void AddNode(int node)
  {
    if (adjacencyList.find(node) != adjacencyList.end()) return;
    adjacencyList[node] = std::vector<std::pair<int, int>>();
  }

  void AddEdge(Edge& edge)
  {
    adjacencyList[edge.fr].push_back({edge.to, edge.wt});
  }

  void PrintAdjacencyList()
  {
    for (std::pair<int, std::vector<std::pair<int, int>>> element : adjacencyList)
    {
      std::cout << element.first << " --> ";
      for (std::pair<int, int> edge : element.second)
      {
        std::cout << edge.first << "[" << edge.second << "]" << " --> ";
      }
      std::cout << "X\n";
    }
  }
};

int main()
{
  std::vector<Edge> edgelist;
  
  edgelist.push_back({1, 2});
  edgelist.push_back({1, 3});
  edgelist.push_back({1, 4});
  edgelist.push_back({2, 3});
  edgelist.push_back({3, 4});
  edgelist.push_back({3, 5});
  edgelist.push_back({4, 5});

  Graph gr(edgelist);

  gr.PrintAdjacencyList();

  return 0;
}